dirty_cat/minhash_encoder.py

```diff
@@ -184,14 +195,48 @@ def transform(self, X):
             X = X[:, 0]
         # Check if first item has str or np.str_ type
         assert isinstance(X[0], str), "ERROR: Input data is not string."
+        if self.minmax_hash:
+            assert self.n_components % 2 == 0,\
+                    "n_components should be even when minmax_hash=True"
+        if self.hashing == 'murmur':
+            assert not(self.minmax_hash),\
+                   "minmax_hash not implemented with murmur"
+        if self.handle_missing not in ['error', '']:
+            template = ("handle_missing should be either 'error' or "
+                        "'', got %s")
+            raise ValueError(template % self.handle_missing)
         X_out = np.zeros((len(X), self.n_components))
+        X = X.ravel()
 
         # TODO Parallel run here
-        for i, x in enumerate(X):
-            if x not in self.hash_dict:
-                self.hash_dict[x] = self.get_hash(x)
+        nan_idx = []
+
+        if self.hashing == 'fast':
+            for i, x in enumerate(X):
+                if isinstance(x, float):
+                    nan_idx.append(i)
+                elif x not in self.hash_dict:
+                    self.hash_dict[x] = self.get_fast_hash(x)
```

Comment:
I think that we need to populate X_out in this loop and remove the one below. The reason being that we can exceed the cache of hash_dict (a bug in the current code).

Reply:
fixed, had to add one more else condition.